external-crates/move/tools/move-package/src/resolution/dependency_graph.rs

```diff
@@ -137,6 +137,9 @@ pub struct Dependency {
     pub subst: Option<PM::Substitution>,
     pub digest: Option<PM::PackageDigest>,
     pub dep_override: PM::DepOverride,
+    /// Original dependency name as defined in parent manifest since it can be different from the
+    /// resolved name. Used for printing user-friendly error messages.
+    pub dep_orig_name: PM::PackageName,
```

Comment:
This commit looks like something great we can ship outright as preparation for compiling against on-chain dependencies, i.e., in separate PR. That would help reduce the footprint / size of the core feature and PR at the end!

Reply:
This issue only happens when `custom_resolve_pkg_name` hook is used and returns something that's different than the manifest so it might not make sense to ship it before the hook code (that PR would be confusing).